### PR TITLE
[BugFix] check config: use_staros when register service from starMgr

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 1]
```diff
@@ -973,7 +973,7 @@ private void transferToMaster(FrontendNodeType oldType) {
 
     // start all daemon threads only running on Master
     private void startMasterOnlyDaemonThreads() {
-        if (Config.integrate_starmgr) {
+        if (Config.integrate_starmgr && Config.use_staros) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:976]: Better to exit if `integrate_starmgr` is true and `use_staros` is false.

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 2]
```diff
@@ -49,20 +49,29 @@ public class StarOSAgent {
 
     public StarOSAgent() {
         serviceId = -1;
-        // check if Config.starmanager_address == FE address
+
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:52]: Remove the blank line.

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 3]
```diff
@@ -49,20 +49,29 @@ public class StarOSAgent {
...
+        workerToId = Maps.newHashMap();
+        workerToBackend = Maps.newHashMap();
+        rwLock = new ReentrantReadWriteLock();
+    }
+
+    public boolean init() {
         if (Config.integrate_starmgr) {
+            if (Config.use_staros == false) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:60]:
```suggestion
            if (!Config.use_staros) {
```

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 4]
```diff
@@ -49,20 +49,29 @@ public class StarOSAgent {
...
+            }
+            // check if Config.starmanager_address == FE address
             String[] starMgrAddr = Config.starmgr_address.split(":");
             if (!starMgrAddr[0].equals("127.0.0.1")) {
                 LOG.warn("Config.starmgr_address not equal 127.0.0.1, it is {}", starMgrAddr[0]);
-                System.exit(-1);
+                return false;
             }
+            client = new StarClient();
+            client.connectServer(Config.starmgr_address);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:70]: Also should init client when `integrate_starmgr` is false.

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java (hunk 3 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:60]: Add error log.

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 20 (35.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|5|20.00%|[851, 852, 992, 993]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|6|15|40.00%|[59, 60, 61, 66, 67, 114, 115, 125, 126]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**